### PR TITLE
backend-hyper: do not disable connection pooling, fix missing ProtocolVersion in ID response

### LINKED ISSUE #132: Connection Pooling is Disabled by Default for the Hyper Implementation
Hello,

by default, the hyper backend disables connection pooling [here](https://github.com/ferristseng/rust-ipfs-api/blob/master/ipfs-api-backend-hyper/src/backend.rs#L56). This is suboptimal for two reasons:
1. Performance: Performing multiple requests with the same client will open a new connection for each request, which is slow and wasteful.
2. Long-running programs using the same client with many requests exhaust the port space of the host. This manifests in errors like this
```
hyper client error `error trying to connect: tcp connect error: Cannot assign requested address (os error 99)`
```

Alternatively, I'm looking at using the `actix` client, but it's not obvious how to enable `rustls` here.

### LINKED ISSUE #134: Latest kubo Removes ProtocolVersion Field in id Response
The latest release of kubo (`v0.22.0`) removes the `ProtocolVersion` field in the `/id` response, see [here](https://github.com/libp2p/go-libp2p/issues/2294) and [here](https://github.com/ipfs/kubo/releases/tag/v0.22.0).

This causes parsing of `IdResponse` structs to fail.

### PR DESCRIPTION
**Pull request by mrd0ll4r**
Fixes #132 
Fixes #134

### CODE REVIEW

File: ipfs-api-backend-hyper/src/backend.rs [hunk 1]
```diff
@@ -52,9 +52,7 @@ macro_rules! impl_default {
 
         impl TryFromUri for HyperBackend<$http_connector> {
             fn build_with_base_uri(base: Uri) -> Self {
-                let client = Builder::default()
-                    .pool_max_idle_per_host(0)
```
Comment (ferristseng) [inline ipfs-api-backend-hyper/src/backend.rs:56]: Hmm, can't remember why I added!

**Comment by ferristseng** [comment]
Looks there is a clippy issue, but I can merge after that. Thanks!

**Comment by iamjpotts** [comment]
The clippy error that's causing the build to fail will be probably be fixed by a rebase. That problem was in master and I had to fix it in my recently merged PR also.

**Comment by mrd0ll4r** [comment]
Rebased, let's see if that fixes it :)

**Comment by mrd0ll4r** [comment]
I also added the `Protocols` field, see [here](https://docs.ipfs.tech/reference/kubo/rpc/#api-v0-id), because why not...
I'm not sure if either `Protocols` or `Addresses` is nullable, but it probably doesn't hurt to guard against it :)

**Pull request merged**